plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -95,26 +97,25 @@ protected IStatus run(IProgressMonitor monitor) {
   };
 
   @VisibleForTesting
-  AnalyticsPingManager(IEclipsePreferences preferences, Display display,
-      ConcurrentLinkedQueue<PingEvent> concurrentLinkedQueue, boolean analyticsEnabled) {
-    this.preferences = preferences;
+  AnalyticsPingManager(String endpointUrl, IEclipsePreferences preferences, Display display,
+      ConcurrentLinkedQueue<PingEvent> concurrentLinkedQueue) {
+    this.endpointUrl = endpointUrl;
+    this.preferences = Preconditions.checkNotNull(preferences);
     this.display = display;
     this.pingEventQueue = concurrentLinkedQueue;
-    this.analyticsEnabled = analyticsEnabled;
   }
 
   public static synchronized AnalyticsPingManager getInstance() {
     if (instance == null) {
       IEclipsePreferences preferences = ConfigurationScope.INSTANCE.getNode(PREFERENCES_PLUGIN_ID);
-      if (preferences == null) {
-        throw new NullPointerException("Preference store cannot be null.");
-      }
-
-      boolean analyticsEnabled = !Platform.inDevelopmentMode() && isTrackingIdDefined();
-      Display display = !analyticsEnabled ? null : PlatformUI.getWorkbench().getDisplay();
+      Display display = PlatformUI.getWorkbench().getDisplay();
```

Comment:
It would be nice to compute this on the fly, with `PlatformUI.getWorkbench()` used as a fallback if `Display.getCurrent() == null` on the off-chance that we're being called on the SWT thread.  And should check that the `PlatformUI.isWorkbenchRunning()` too.